tests/python/pants_test/engine/test_build_files.py

```diff
@@ -200,9 +200,12 @@ def create_json(self):
   def _populate(self, scheduler, address):
     """Perform an ExecutionRequest to parse the given Address into a Struct."""
     request = scheduler.execution_request([TestTable().constraint()], [address])
-    root_entries = scheduler.execute(request).root_products
-    self.assertEqual(1, len(root_entries))
-    return request, root_entries[0][1]
+    returns, throws = scheduler.execute(request)
+    if returns:
+      state = returns[0][1]
+    else:
+      state = throws[0][1]
```

Comment:
Q:
1. is it possible to have both returns and throws? in which case the logic seems to ignore throws.
2. is there certain expectation on `execute`'s return value, e.g. `len(throws) + len(returns) = len(request.roots)`?